beginner_source/hybrid_frontend/introduction_to_hybrid_frontend_tutorial.py

```diff
@@ -0,0 +1,526 @@
+# a scalar tensor for ``k``.
+#
+
+# Takes parallel distances and labels tensors and returns the sorted labels
+def getSortedLabels(distances, labels, k):
+    # Sort distances
+    sorted_dists, indices = torch.sort(distances)
+    # Reorder labels according to sorted indices
+    sorted_labels = torch.gather(labels, 0, indices)
+    # Return top-k sorted_labels
+    return sorted_labels[:k]
+
+
+########## Example ##########
+# Non-compiled
+print("Sorted labels example:")
+ds = torch.tensor([6, 8, 3, 5, 9])
+ls = torch.tensor([0, 0, 2, 1, 2])
+k = torch.tensor(3)
+sorted_labels = getSortedLabels(ds, ls, k)
+print("Non-compiled output:", sorted_labels)
+
+# Compiled
+getSortedLabels_traced = trace(ds, ls, k)(getSortedLabels)
+sorted_labels_traced = getSortedLabels_traced(ds, ls, k)
+print("Compiled output:", sorted_labels_traced)
+print("getSortedLabels_traced.graph:", getSortedLabels_traced.graph)
+
+
+######################################################################
+# getNeighbors
+# ^^^^^^^^^^^^
+#
+# The ``getNeighbors`` function takes a ``train_set``, a
+# ``test_instance``, an empty ``distances`` tensor, an empty ``labels``
+# tensor, and a ``k`` value. The ``distances`` and ``labels`` tensors must
+# have same length as ``train_set``. It returns a list of the sorted
+# *k-nearest* neighbors (as class labels) to the ``test_instance``. It
+# does this by iterating over every instance in the ``train_set``,
+# recording the distance between the train instance and the
+# ``test_instance``, and using our ``getSortedLabels`` function to sort
+# the corresbonding labels based on these distances.
+#
+# --------------
+#
+# **Script mode**
+#
+# Notice that this function contains a data-dependent control flow (our
+# *for* loop depends on the data size). Because of this, we must compile
+# with **script** mode. We indicate the use of this mode by adding a
+# ``@script`` decorator. Note that we can call our traced
+# ``getDistance_traced``, and ``getSortedLabels_traced`` functions from
+# our script function.
+#
+# Finally, you may be wondering why we use the ``addEntries`` function to
+# add to our ``distances`` and ``labels`` tensors instead of simply:
+#
+# ::
+#
+#     distances[i] = dist
+#     labels[i] = label
+#
+#
+# Remember that only a subset of Python is supported in **script** mode.
+# One requirement is that the only expressions allowed on the left side of
+# an assignment operator (=) are variable names and starred expressions.
+# So, one way to circumvent this is to implement pure Python helper
+# functions such as ``addEntries``.
+#
+# Also, notice in the graph output that the loop is captured
+# (``prim::Loop``). The **script** mode enables us to convert the Python
+# AST directly to graph mode, rather than tracing an example input through
+# the control flow and simply capturing the unrolled loop.
+#
+
+# Add elements to distances and labels tensors at index i
+def addEntries(i, d, l, distances, labels):
```

Comment:
I am not sure this is the best example, unless you follow it up with an example that shows how to eliminate the Python. As it stands, it is not possible to call `.save` on this model because it has a call.